Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -5381,7 +5381,19 @@ void WebPageProxy::findStringMatches(const String& string, OptionSet<FindOptions
 
 void WebPageProxy::findString(const String& string, OptionSet<FindOptions> options, unsigned maxMatchCount, CompletionHandler<void(bool)>&& callbackFunction)
 {
-    auto sendFindStringMessage = [&]<typename M>(M&& message, auto&& completionHandler)
+    if (!m_browsingContextGroup->hasRemotePages(*this)) {
+        auto completionHandler = [protectedThis = Ref { *this }, string, callbackFunction = WTFMove(callbackFunction)](std::optional<FrameIdentifier> frameID, Vector<IntRect>&& matchRects, uint32_t matchCount, int32_t matchIndex, bool didWrap) mutable {
+            if (!frameID)
+                protectedThis->findClient().didFailToFindString(protectedThis.ptr(), string);
```

Comment:
This is duplicate code with the lambda in FindStringCallbackAggregator destructor.  Can it be shared?

Reply:
The lambda capture and parameters are different to use aggregated results in `~FindStringCallbackAggregator()`. I could create a nested lambda to avoid duplicating the logic inside the lambda, but that didn’t seem much better to me.